### PR TITLE
Feat/msdk 219 get transaction history

### PR DESCRIPTION
**Pull request by shuffledex**

### CODE REVIEW

File: src/types/index.ts [hunk 1]
```diff
@@ -222,6 +223,18 @@ export interface IdentityWithClaims {
   claims: ClaimData[];
 }
 
+export interface ExtrinsicData {
```
Comment (monitz87) [inline src/types/index.ts:226]: Why are these values optional?

File: src/types/index.ts [hunk 2]
```diff
@@ -3,6 +3,7 @@ import { IKeyringPair, TypeDef } from '@polkadot/types/types';
 import BigNumber from 'bignumber.js';
 
 import { Identity } from '~/api/entities';
+import { TxTag } from '~/polkadot';
```
Comment (monitz87) [inline src/types/index.ts:6]:
```suggestion
import { TxTag } from 'polymesh-types/types';
```

File: src/utils/__tests__/index.ts [hunk 3]
```diff
@@ -2099,6 +2102,28 @@ describe('txTagToProtocolOp', () => {
   });
 });
 
+describe('txTagToExtrinsicIdentifier and extrinsicIdentifierToTxTag', () => {
```
Comment (monitz87) [inline src/utils/__tests__/index.ts:2105]: Use examples where both the module and the call names have more than one word in them

File: src/Polymesh.ts [hunk 4]
```diff
@@ -665,15 +665,16 @@ export class Polymesh {
...
    *
    * @param opts.address - account that signed the transaction
+   * @param opts.tag - tag associated with the transaction
    * @param opts.success - whether the transaction was successful or not
    * @param opts.size - page size
    * @param opts.start - page offset
    */
   public async getTransactionHistory(
-    opts: {
+    filters: {
```
Comment (monitz87) [inline src/Polymesh.ts:674]: Change this back to opts or change the comments to reflect this new name

File: src/Polymesh.ts [hunk 5]
```diff
@@ -682,20 +683,23 @@ export class Polymesh {
...
 
-    let extrinsicIdentifier;
-    if (txTag) {
-      extrinsicIdentifier = txTagToExtrinsicIdentifier(txTag);
+    let moduleId;
+    let callId;
+    if (tag) {
+      const extrinsicIdentifier = txTagToExtrinsicIdentifier(tag);
+      moduleId = extrinsicIdentifier.moduleId;
+      callId = extrinsicIdentifier.callId;
```
Comment (monitz87) [inline src/Polymesh.ts:693]: You can just destructure here

File: src/types/index.ts [hunk 6]
```diff
@@ -224,15 +224,15 @@ export interface IdentityWithClaims {
 }
 
 export interface ExtrinsicData {
-  blockId?: number;
-  extrinsicIdx?: number;
+  blockId: number;
+  extrinsicIdx: number;
   address?: string | null;
```
Comment (monitz87) [inline src/types/index.ts:229]: Why is this value still optional (or null)?

Comment (shuffledex) [reply src/types/index.ts:229]: address can be null if the user searches for system transactions. @VictorVicente can add more details about it.

Comment (monitz87) [reply src/types/index.ts:229]: Then have it just be `string | null`. Doesn't make sense for it to be undefined as well

**Pull request merged**